src/DataStructures/ModalVector.hpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/PointerVector.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <blaze/math/expressions/DVecMapExpr.h>
+// IWYU pragma: no_include <blaze/math/typetraits/IsVector.h>
+// IWYU pragma: no_include "DataStructures/DataVector.hpp"
+// IWYU pragma: no_forward_declare ConstantExpressions_detail::pow
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A class for storing spectral coefficients on a mesh.
+ *
+ * \details A ModalVector holds an array of spectral coefficients, and can be
+ * either owning (the array is deleted when the ModalVector goes out of scope)
+ * or non-owning, meaning it just has a pointer to an array.
+ *
+ * Only basic mathematical operations are supported with ModalVectors. In
+ * addition to addition, subtraction, multiplication, division, there
+ * are the following element-wise operations:
+ *
+ * - max
+ * - min
+ */
+class ModalVector : public VectorImpl<double, ModalVector> {
+ public:
+  using VectorImpl<double, ModalVector>::operator=;
+  using VectorImpl<double, ModalVector>::VectorImpl;
+
+  MAKE_EXPRESSION_MATH_ASSIGN_PV(+=, ModalVector)
+  MAKE_EXPRESSION_MATH_ASSIGN_PV(-=, ModalVector)
+};
+
+namespace blaze {
+template <>
+struct IsVector<ModalVector> : std::true_type {};
+template <>
+struct TransposeFlag<ModalVector> : BoolConstant<ModalVector::transpose_flag> {
+};
+BLAZE_TRAIT_SPEC_BINTRAIT(ModalVector, AddTrait);
+BLAZE_TRAIT_SPEC_BINTRAIT(ModalVector, SubTrait);
+BLAZE_TRAIT_COMPATIBLE_BINTRAIT(ModalVector, DataVector, AddTrait);
```

Comment:
I think we don't want to be able to mix any operations between `ModalVector` and `DataVector`, right @kidder ?

Reply:
No, I don't know what that even means